routing/routing_helpers.cpp

```diff
@@ -33,6 +33,16 @@ void ReconstructRoute(IDirectionsEngine * engine, IRoadGraph const & graph,
   if (engine)
     engine->Generate(graph, path, times, turnsDir, junctions, trafficSegs, cancellable);
 
+  if (cancellable.IsCancelled())
+    return;
+
+  // In case of any errors in IDirectionsEngine::Generate() |junctions| is empty.
+  if (junctions.empty())
```

Comment:
In case path.size() == 1, it's not an error. But in case of ReconstructPath() it is an error.
Do you think it's better to remove assert (LOG(LERROR,...) from here?